### ISSUE TITLE
Calendar doesn't read all events from Huawei/Xiaomi phones

### ISSUE THREAD

**Issue by hufman** [BimmerGestalt/AAIdrive#508]
Split from #279 @jezikk82 and @berseker report that the app can't see all calendar events.
It seems to be that most calendars are [marked as not visible](https://github.com/BimmerGestalt/AAIdrive/issues/279#issuecomment-1013014045)

**Comment by hufman** [comment]
@jezikk82  Try out [this branch](https://github.com/BimmerGestalt/AAIdrive/compare/calendar_ignore_visibility) which ignores calendar visibility entirely. I could probably add this as an advanced setting if it works for you.

**Comment by jezikk82** [comment]
@hufman Affected are all phone based on miui/emui so Xiaomi/Redmi and others as well.

The branch you gave me to test works fine, but playing around I found that on Xiaomi/Redme phones you separately subscribe the visibility of calendars. I don't know how but when I change visibility in Mi Calendar app it's changing the visibility in Android and in AAID but not in GCal app. The other way round in GCal I can have different calendars shown I in app I see this events but it doesn't change Android subscription calendars. 

e.g. In Samsung GCal visible calendars are replicated to Android (and AAID) visibility. 

To summarise: Making the switch to ignore visibility should partially solve the problem. Partially, because you wouldn't have the ability to choose which calendars present in car. However, better is to have all than nothing.

**Comment by hufman** [comment]
Very curious, that's an interesting clue!
I'll push up the option, which will provide a workaround at least.
Thank you for your patience in testing!

**Comment by berseker** [comment]
Ahah i was going to test It on my Xiaomi in a few hours but the bug have been already closed 😅 in any case i'll do a report anyway

**Comment by hufman** [comment]
I'd welcome your feedback, please! ❤️

**Comment by berseker** [comment]
actually I do not know If i do something wrong but calendar appointments does not show up for me, I see only the "Festività in Italia" dates  (Italian Holidays).. I used also the latest main build of AAiDrive.. maybe it would be useful to have other Xiaomi users feedback on this subject .. I have installed both MiCalendar & Google Calendar, if I set up an appointment in one or in the other, they show up in both
![image](https://user-images.githubusercontent.com/1159431/150637007-7cb62e71-99a5-4f0b-bb19-d0c04754b072.png)

**Comment by hufman** [comment]
The latest main build includes an Advanced Setting to show events from hidden calendars, does that show anything more? Though your calendars don't appear hidden.
What events are you expecting to see instead?

**Comment by jezikk82** [comment]
Try to disable checkbox to hide events without details. Maybe thats' the reason you don't see many events.

**Comment by berseker** [comment]
> Try to disable checkbox to hide events without details. Maybe thats' the reason you don't see many events.

I confirm that disabling the option finally solved my issue. I think that by error I did not activate this option correctly.. my bad .. thank you for your support!
